src/App.tsx

```diff
@@ -1,39 +1,67 @@
 import React from 'react';
 import './App.scss';
+import CircularProgress from '@mui/material/CircularProgress';
+import Box from '@mui/material/Box';
+import { Clock } from './components/Clock';
 
-function getRandomName(): string {
-  const value = Date.now().toString().slice(-4);
+type State = {
+  clockName: string;
+  hasClock: boolean;
+};
 
-  return `Clock-${value}`;
-}
+export class App extends React.Component<{}, State> {
+  state = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+  timerId = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount(): void {
+    document.addEventListener('contextmenu', this.clickOrContextHandler);
+    document.addEventListener('click', this.clickOrContextHandler);
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: this.getRandomName() });
+    }, 3300);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentWillUnmount(): void {
+    document.removeEventListener('contextmenu', this.clickOrContextHandler);
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  getRandomName = () => {
+    const value = Date.now().toString().slice(-4);
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+    return `Clock-${value}`;
+  };
 
-        {' time is '}
+  clickOrContextHandler = (event: MouseEvent) => {
```

Comment:
checklist p3
and I think that you should separate it to two diferent functions

```suggestion
  handleClickOrContext = (event: MouseEvent) => {
```
or better:

```suggestion
  handleClick = (event: MouseEvent) => {...}
  handleContext = (event: MouseEvent) => {...}
```

Reply:
fix these comments, please, rename - checklist p3 
and split logic